### ISSUE TITLE
F401 false positive when importing in a type-checking block

### ISSUE THREAD

**Issue by alexandermalyga** [astral-sh/ruff#12780]
A false positive `F401` error is raised when the import is in a type-checking block and used as a generic parameter.

foo.py
```python
class Foo: ...
```

bar.py
```python
class Bar[T]: ...
```

baz.py
```python
from bar import Bar
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from foo import Foo

class Baz(Bar["Foo"]): ...
```

raises

```
baz.py:5:21: F401 [*] `foo.Foo` imported but unused
  |
4 | if TYPE_CHECKING:
5 |     from foo import Foo
  |                     ^^^ F401
6 | 
7 | class Baz(Bar["Foo"]):
  |
  = help: Remove unused import: `foo.Foo`

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

Command and Ruff version:

`ruff check baz.py --isolated`
Version 0.5.7

**Comment by charliermarsh** [comment]
We don't treat arbitrary subscripts as generics, since they aren't known to be generics at present (i.e., that could be a dictionary access). But you can mark `bar.Bar` as a generic via [`extend-generics`](https://docs.astral.sh/ruff/settings/#lint_pyflakes_extend-generics):

```toml
[tool.ruff.lint.pyflakes]
extend-generics = ["bar.Bar"]
```

**Comment by AlexWaygood** [comment]
Thanks for the report! Closing as a duplicate of #9298, however :-)
